packages/vite/src/types.ts

```diff
@@ -77,4 +77,10 @@ export interface VitePluginOptions {
    * @deprecated
    */
   force?: boolean
+
+  /**
+   * Set to true to read all frames/pages of an animated image (GIF, WebP, TIFF), equivalent of setting pages to -1.
+   * @default false
+   */
+  animated: boolean
```

Comment:
the option @neil3000 suggested was a per-image option (aka a directive) passed as a query string: https://github.com/JonasKruckenberg/imagetools/blob/main/docs/directives.md.  

This is a per-project option, which is probably not what we want. We can use `defaultDirectives` to turn it entirely on or off for a project: https://github.com/JonasKruckenberg/imagetools/tree/main/packages/vite#defaultdirectives

Reply:
Indeed. I think it's better for this parameter to be a per-image option, as most of the time the developer know if the image he included on the site is animated or not.

If it's dynamic content (like user generated content) they can choose to implement their own logic :)

Reply:
thank you for the feedback. I will look into this the next days. I found there are also some things I got wrong for the tests, so it might be better I close this PR for now and will come up with a new one when I worked myself into the logic of this repo and managed to implement this feature and the tests.